architecture/2024-11-ucp-component-threat-model.md

```diff
@@ -0,0 +1,291 @@
+   "name": "RequestPipelineStart",
+   "scope": "System.Net.Http.HttpClient.local.LogicalHandler",
+   "serviceName": "deployment.engine",
+   "version": "0.38\u002B1b2449770894506793f865dc7ecc9ed8e6df0d93",
+   "hostName": "bicep-de-6b86cc859f-tt2r8",
+   "traceId": "a0d750fc4e5d35bd1c615d87c1261f72",
+   "spanId": "5b624ecddd0b19c2",
+   "HttpMethod": "GET",
+   "Uri": "https://ucp.radius-system/apis/api.ucp.dev/v1alpha3/planes/radius/local/resourceGroups/default/providers/Applications.Core/containers/demo?api-version=2023-10-01-preview"
+   },
+   {
+   "timestamp": "2024-11-03T14:06:54.684Z",
+   "severity": "INFO",
+   "message": "Sending HTTP request GET https://ucp.radius-system/apis/api.ucp.dev/v1alpha3/planes/radius/local/resourceGroups/default/providers/Applications.Core/containers/demo?api-version=2023-10-01-preview",
+   "name": "RequestStart",
+   "scope": "System.Net.Http.HttpClient.local.ClientHandler",
+   "serviceName": "deployment.engine",
+   "version": "0.38\u002B1b2449770894506793f865dc7ecc9ed8e6df0d93",
+   "hostName": "bicep-de-6b86cc859f-tt2r8",
+   "traceId": "a0d750fc4e5d35bd1c615d87c1261f72",
+   "spanId": "5b624ecddd0b19c2",
+   "HttpMethod": "GET",
+   "Uri": "https://ucp.radius-system/apis/api.ucp.dev/v1alpha3/planes/radius/local/resourceGroups/default/providers/Applications.Core/containers/demo?api-version=2023-10-01-preview"
+   }
+   ```
+
+2. **Health Check Probes**: Kubernetes itself can act as a client by performing health and readiness checks on the Universal Control Plane.
+
+3. **Metrics Scrapers**: If metrics are enabled, Prometheus or other monitoring tools can scrape metrics from the Universal Control Plane.
+
+## Trust Boundaries
+
+### UCP Container runs as Non-Root
+
+As of November 24, 2024, the UCP container runs as a non-root user. This security measure helps to limit the potential impact of a container compromise by restricting the privileges available to the container. Running containers as non-root users is a best practice that enhances the overall security posture of the system. For more details, refer to the [Dockerfile](https://github.com/radius-project/radius/blob/96504063bbab8c0ee53b0955a9647cf00c7f5fae/deploy/images/ucpd/Dockerfile#L14).
+
+### Trust Model of UCP Clients
+
+UCP is a [Kubernetes API Service](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/apiserver-aggregation/) that aims to extend Kubernetes with its APIs and capabilities. So anything that is sent to an API path that starts with `/apis/api.ucp.dev/v1alpha3` gets sent to the UCP through the Kubernetes API Server. This means that for external clients like Radius CLI, we rely on Kubernetes authentication and don't provide any additional authentication mechanism.
+
+![CLI - Kubernetes API Server - UPC](./2024-11-ucp-component-threat-model/cli-kubernetes-api-server-ucp.png)
```

Comment:
I'm glad you added this. I think if we frame this in the lens of "trust boundaries", we could say that requests coming into UCP from a client are originating from an **authorized user**.